src/shims/env.rs

```diff
@@ -308,8 +310,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         Ok(Scalar::ptr_null(&*this.tcx))
     }
 
+    #[allow(non_snake_case)]
+    fn GetCurrentDirectoryW(
+        &mut self,
+        size_op: OpTy<'tcx, Tag>, // DWORD
+        buf_op: OpTy<'tcx, Tag>,  // LPTSTR
+    ) -> InterpResult<'tcx, u32> {
+        let this = self.eval_context_mut();
+        this.assert_target_os("windows", "GetCurrentDirectoryW");
+
+        this.check_no_isolation("GetCurrentDirectoryW")?;
+
+        let size = u64::from(this.read_scalar(size_op)?.to_u32()?);
+        let buf = this.read_scalar(buf_op)?.not_undef()?;
+
+        // If we cannot get the current directory, we return 0
+        match env::current_dir() {
+            Ok(cwd) => {
+                let len = this.write_path_to_wide_str(&cwd, buf, size)?.1;
+                return Ok(u32::try_from(len).unwrap())
```

Comment:
This entirely ignores whether the buffer was big enough to hold the string (`.0` component of what `write_path_to_wide_str` returns). Is that how Windows operates?

The docs say

> If the buffer that is pointed to by lpBuffer is not large enough, the return value specifies the required size of the buffer, in characters, including the null-terminating character.

which is not what the function here does. This is the same as the "get env var" shim, so it is worth factoring out the common code into a (local, for now) helper function.

Reply:
Currently, `write_path_to_wide_str` calls to `write_os_str_to_wide_str`, which returns exactly what `write_path_to_wide_str` wants. I thought it is not necessary to check the boolean return value at all.
```rust
/// Helper function to write an OsStr as a 0x0000-terminated u16-sequence, which is what
/// the Windows APIs usually handle. This function returns `Ok((false, length))` without trying
/// to write if `size` is not large enough to fit the contents of `os_string` plus a null
/// terminator. It returns `Ok((true, length))` if the writing process was successful. The
/// string length returned does not include the null terminator.
    fn write_os_str_to_wide_str(
```

Reply:
I'll fix this in the upcoming commit